src/pip/_internal/utils/misc.py

```diff
@@ -1140,14 +1140,26 @@ def build_url_from_netloc(netloc, scheme='https'):
     return '{}://{}'.format(scheme, netloc)
 
 
-def netloc_has_port(netloc):
-    # type: (str) -> bool
+def parse_netloc(netloc):
+    # type: (str) -> Tuple[str, Optional[int]]
     """
-    Return whether the netloc has a port part.
+    Return the host-port pair from a netloc.
     """
     url = build_url_from_netloc(netloc)
     parsed = urllib_parse.urlparse(url)
-    return bool(parsed.port)
+    return parsed.hostname, parsed.port
+
+
+def build_netloc(host, port):
+    # type: (str, Optional[int]) -> str
+    """
+    Build a netloc from a host-port pair
+    """
+    if host.count(':') >= 2 and '@' not in host and '[' not in host:
+        # It must be a bare IPv6 address, so wrap it with brackets.
+        host = '[{}]'.format(host)
```

Comment:
Since this is going to be for the `--trusted-host` command-line option rather than for inclusion in a URL, I think we only want to be including brackets if a port is present.

Reply:
I am afraid not, the URL is also being used in session adapters, where a valid URL(prefix) should be present.

Reply:
This is the only place in the code using the return value of `build_netloc()`: https://github.com/pypa/pip/blob/9b9d7b7d5dc87efb7a0ed44c07749d0577e35745/src/pip/_internal/build_env.py#L194-L195

It's only for constructing a string to pass as `--trusted-host` in the CLI.

Reply:
The session adapter code is using a different function / code path (`build_url_from_netloc()`).

Reply:
Oh, i thought it was `build_url_from_netloc()`, will change.

Reply:
No, that code is fine. My comments were just about the new function you added..